### PR TITLE
HARP-11448: Adds actual deployment step to github actions publishing …

### PR DESCRIPTION
**Pull request by FraukeF**
…workflow

Signed-off-by: Frauke Fritz <frauke.fritz@here.com>

Thank you for contributing to harp.gl!

Before requesting a pull request, please remember to check the following documents:
* [contribution guidelines](https://github.com/heremaps/harp.gl/blob/master/CONTRIBUTING.md)
* [coding style](https://github.com/heremaps/harp.gl/blob/master/CODINGSTYLE.md)

If you are adding new functionality we would highly appreciate if you can describe what is the capability you are adding and even better if you can add some examples. Please also remember to add tests for it.

# CI Check

Our bots will check whether your PR can be directly integrated into the mainline. We have some internal integration tests running on the background, our bots will inform you of the next steps and someone from our team will take a look and help if needed!

And please do not forget to sign-off your commit! You can read more about DCO [here](https://julien.ponge.org/blog/developer-certificate-of-origin-versus-contributor-license-agreements/). But, in short, you just need to use `git commit -s` or append `--signoff` when you are committing to the repo.

Happy contributing!

### CODE REVIEW

File: .github/workflows/deploy_to_npm.yaml [hunk 1]
```diff
@@ -1,10 +1,11 @@
-name: Publish Packages
+name: Build & Deploy
 
 on:
     workflow_dispatch:
     push:
         branches:
             - release
+            - master
```
Comment (nzjony) [inline .github/workflows/deploy_to_npm.yaml:8]: this will also release on master branch?

Comment (FraukeF) [reply .github/workflows/deploy_to_npm.yaml:8]: yes, but there is a filter in the deployment step, so it does not publish npms, bu only the documentation and examples, it matches the old workflow

**Review by nzjony — APPROVED** [review]

**Comment by FraukeF** [comment]
rebase please

**Comment by FraukeF** [comment]
retest please

**Review by gairik — APPROVED** [review]

**Pull request merged**